src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -96,6 +97,12 @@
         StringLexer("greater_than_or_equal", ">=", CodeSegment),
         StringLexer("less_than_or_equal", "<=", CodeSegment),
         RegexLexer("newline", r"\r\n|\n", NewlineSegment),
+        RegexLexer(
+            "ignore",
+            r"(^\s*\\)[^\n]*",
+            IgnoreSegment,
+            segment_kwargs={"trim_start": (r"^\s*\\")},
+        ),
```

Comment:
Quick question: when the `noqa` tag is included, does `sqlfluff` not still parse the code in the scope of the `noqa`?

Reply:
It still parses the code. The use of `noqa` tells it to discard any lint warnings for the affected lines of code. I'm not sure if this also filters out parse errors.

Reply:
You can ignore rule errors (e.g. with `-- noqa: L007, L006`) or parsing errors (e.g. with `-- noqa: PRS`). You can’t ignore lever errors (there is `no -- noqa: LEX` but maybe there should be?). More info here: https://docs.sqlfluff.com/en/stable/configuration.html#ignoring-errors-files

When ignoring parsing errors it seems pretty good at picking up from where it left off so the whole statement is not ignored but the rest is parsed (with just an `UnparsableSegment` in the middle) and other syntax errors still highlighted. More info here: https://docs.sqlfluff.com/en/stable/architecture.html#stage-3-the-parser

Reply:
If the -- noqa: PRS or the like enables the user to _effectively_ skip the code whereby does not prevent subsequent lint/fixes, this would accomplish the goal.  This is "user-directed" identity function application (the way I framed it).  That would be great (transparent to the user).

In my experience of the parser, when `UnparsableSegment` is encountered, all of the output fails.  i.e., no linting.

It may still be useful to have such a segment, i.e.,`UnparsableSegment` that throws an Exception.  What does `-- noqa: PRS` construct?